### PR TITLE
update L1 data fee for curie

### PR DESCRIPTION
**Pull request by DreamWuGit**
### Description

- [x] add feature `l1_fee_curie` switch for curie fork
- [x] buss mapping update transaction to get new fields 
- [x] update circuit(L1fee gadget) for new fields operations and apply new formula
  - [x]  begin_tx 
  - [x] end_tx
  - [x] TxL1FeeGadget
- [x] testing stuffs
### Issue Link

[_link issue here_]

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by DreamWuGit** [comment]
mark ready for ci checking purpose

File: bus-mapping/src/circuit_input_builder/transaction.rs [hunk 1]
```diff
@@ -516,11 +516,11 @@ impl TxL1Fee {
                 (tx_l1_fee % TX_L1_FEE_PRECISION as u128) as u64,
             )
         }
-
+        
         #[cfg(feature = "l1_fee_curie")]
         {
             // "commitScalar * l1BaseFee + blobScalar * _data.length * l1BlobBaseFee",
-            let tx_l1_fee = self.commit_scalar as u128 * self.l1_blob_basefee as u128
+            let tx_l1_fee = self.commit_scalar as u128 * self.base_fee as u128
```
Comment (lispc) [inline bus-mapping/src/circuit_input_builder/transaction.rs:525]: remember to update comments.  We should add a link to l2geth code as above.

File: bus-mapping/src/l2_predeployed.rs [hunk 2]
```diff
@@ -36,8 +36,11 @@ pub mod l1_gas_price_oracle {
     pub static SCALAR_SLOT: LazyLock<U256> = LazyLock::new(|| U256::from(3));
 
     /// THe following 3 slots plus `BASE_FEE_SLOT` will be used for l1 fee after curie fork
+    /// L1 BlobBaseFee slot in L1GasPriceOracle after Curie fork
     pub static L1_BLOB_BASEFEE_SLOT: LazyLock<U256> = LazyLock::new(|| U256::from(5));
+    /// L1 commitScalar slot in L1GasPriceOracle after Curie fork
     pub static COMMIT_SCALAR_SLOT: LazyLock<U256> = LazyLock::new(|| U256::from(6));
+    /// L1 commitScalar slot in L1GasPriceOracle after Curie forks
```
Comment (lispc) [inline bus-mapping/src/l2_predeployed.rs]: "Curie forks".   the above lines are "after Curie fork".  And line 43 should be blobScalar ? not commitScalar?

File: zkevm-circuits/src/evm_circuit/util/common_gadget.rs [hunk 3]
```diff
@@ -841,6 +841,12 @@ impl<F: Field> TransferToGadget<F> {
         } else {
             (0.into(), 0.into())
         };
+
+        #[cfg(feature = "l1_fee_curie")]
+        println!("l1_fee_curie enabled");
+        #[cfg(not(feature = "l1_fee_curie"))]
+        println!("l1_fee_curie not enabled");
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/util/common_gadget.rs]: remove to switch to log::debug or remove finally

File: bus-mapping/src/circuit_input_builder/transaction.rs (hunk 1 above)
Comment (DreamWuGit) [reply bus-mapping/src/circuit_input_builder/transaction.rs:525]: yes, will update when do last clean up

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs [hunk 4]
```diff
@@ -31,33 +31,64 @@ pub(crate) struct TxL1FeeGadget<F> {
     fee_overhead_word: U64Word<F>,
     /// Current value of L1 fee scalar
     fee_scalar_word: U64Word<F>,
-    /// Committed value of L1 base fee
+    #[cfg(feature = "l1_fee_curie")]
+    /// Current value of L1 blob base fee
+    l1_blob_basefee_word: U64Word<F>,
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:40]: need to check with peter whether they really in range u64.

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs [hunk 5]
```diff
@@ -78,13 +109,46 @@ impl<F: Field> TxL1FeeGadget<F> {
...
         );
 
+        // TODO: check if can really reuse base_fee_slot read rw above for curie
+        // now try to skip it.
+        // for curie hard fork
+        #[cfg(feature = "l1_fee_curie")]
+        // Read l1blob_baseFee_committed
+        cb.account_storage_read(
+            l1_fee_address.expr(),
+            l1_blob_baseFee,
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs]: i guess "baseFee" is not a good rust naming

**Comment by lispc** [comment]
if scroll-prover test passes, i will merge this. But later i think we should add some unit tests in repo.

File: eth-types/src/geth_types.rs [hunk 6]
```diff
@@ -158,6 +158,48 @@ pub fn get_rlp_unsigned(tx: &crate::Transaction) -> Vec<u8> {
     }
 }
 
+/// Get the RLP signed bytes
+/// NOTE: currently only used by curie upgrade calculating l1fee
+pub fn get_rlp_signed(tx: &crate::Transaction) -> Vec<u8> {
```
Comment (lispc) [inline eth-types/src/geth_types.rs]: @kunxian-xia is there any better method to do this?

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs [hunk 7]
```diff
@@ -158,8 +289,32 @@ impl<F: Field> TxL1FeeGadget<F> {
...
+            0.expr()
+        } else {
+            tx_data_gas_cost + TX_L1_COMMIT_EXTRA_COST.expr() + fee_overhead
+        };
+
+        // TODO: new formula for curie
+        #[cfg(feature = "l1_fee_curie")]
+            cb.require_equal(
+            "commitScalar * l1BaseFee + blobScalar * _data.length * l1BlobBaseFee == tx_l1_fee * 10e9 + remainder",
+            commit_scalar * base_fee + blob_scalar * tx_signed_length * l1_blob_basefee,
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:328]: Is this sound? Anywhere we range check remainder?

File: zkevm-circuits/src/evm_circuit/util/common_gadget.rs (hunk 3 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget.rs]: removed

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs (hunk 5 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs]: changed to "basefee"

**Comment by DreamWuGit** [comment]
> if scroll-prover test passes, i will merge this. But later i think we should add some unit tests in repo.

wow, test have passed!!

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs (hunk 7 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:328]: there is no range check remainder for old l1 fee such field,  should be a bug, will file an issue to trace this and fix later.

Comment (lispc) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:328]: @noel2004 can have a look

Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:328]: filed bug https://github.com/scroll-tech/zkevm-circuits/issues/1307

File: eth-types/src/geth_types.rs (hunk 6 above)
Comment (kunxian-xia) [reply eth-types/src/geth_types.rs]: I think we can just use `tx.rlp()` instead.

File: bus-mapping/src/circuit_input_builder/transaction.rs (hunk 1 above)
Comment (DreamWuGit) [reply bus-mapping/src/circuit_input_builder/transaction.rs:525]: added the new link

File: eth-types/src/geth_types.rs (hunk 6 above)
Comment (DreamWuGit) [reply eth-types/src/geth_types.rs]: seems ok , will have a test.

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs (hunk 7 above)
Comment (noel2004) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:328]: Have purposed a fixing in https://github.com/scroll-tech/zkevm-circuits/pull/1308/files

File: eth-types/src/geth_types.rs (hunk 6 above)
Comment (DreamWuGit) [reply eth-types/src/geth_types.rs]: tested pass， will remove this new helper

Comment (DreamWuGit) [reply eth-types/src/geth_types.rs]: removed.

File: zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs (hunk 7 above)
Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs:328]: merged into this pr .

**Pull request merged**